### DISCUSSION TITLE
Pass visual selection to plugin `keys` mapping

### DISCUSSION THREAD

**Discussion by sebiwi** [LazyVim/LazyVim#4579] [Q&A] [answered]
Hello, I'm trying to pass a visual selection to a `keys` mapping for a specific plugin. In this case, it's Diffview's `DiffviewFileHistory` command.

Here is my current configuration:

```
return {
  {
    "NeogitOrg/neogit",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "sindrets/diffview.nvim",
      "nvim-telescope/telescope.nvim", -- optional
    },
    config = true,
    keys = {
      {
        "<LEADER>gg",
        "<cmd>Neogit<cr>",
        desc = "Neogit",
      },
      {
        "<LEADER>gf",
        "<cmd>DiffviewFileHistory %<cr>",
        desc = "Diffview Current File History",
      },
      {
        mode = "v",
        "<LEADER>gf",
        ":DiffviewFileHistory<CR>",
        desc = "Diffview Current Selection History",
      },
    },
  },
}
```

The third mapping does what I want, but it makes the command window appear for a split second. I've seen this pattern on other parts of LazyVim (e.g. [here](https://www.lazyvim.org/plugins/editor#gitsignsnvim)).

I'd like to make this work without the command window showing. I tried using `<cmd>` before the command instead of `:` as the `Diffview Current File History` mapping above the one that works in visual mode, without success (the range doesn't get passed to the command).

Am I missing something?

**Comment by dpetka2001** [comment, marked as answer]
Replace the `rhs` with
```lua
        function()
          local start_line, end_line =
            math.min(vim.fn.line("."), vim.fn.line("v")), math.max(vim.fn.line("."), vim.fn.line("v"))
          vim.cmd(string.format("%d,%d DiffviewFileHistory", start_line, end_line))
        end,
```
